src/frontend/contexts/categories-context.js

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect, createContext, useContext } from "react";
+import axios from "axios";
+
+const CategoriesContext = createContext();
+
+const CategoriesProvider = ({ children }) => {
+  const [categories, setCategories] = useState([]);
+  
+  const categoryNames = categories.map((category) => category.name);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const { data } = await axios.get("/api/categories");
+        setCategories(data.categories);
+      } catch (err) {
+        console.error(err.message);
+      }
+    })();
+  }, []);
+
+  return (
+    <CategoriesContext.Provider value={{ categories, categoryNames }}>
+      {children}
+    </CategoriesContext.Provider>
+  );
+};
+
+const useCategories = () => useContext(CategoriesContext);
+
```

Comment:
You can use this condition when using useContext, so if anytime you get into an error while using it, you can catch it.
`const useCategories = () => {
  const context = useCategories(CategoriesContext);

  if (context === undefined) {
    throw new Error("useCategories must be used within a theme provider");
  }
  return context;
};`

Reply:
Damn! Never knew about this. Thank you for this! 👌